token-group/example/src/processor.rs

```diff
@@ -0,0 +1,225 @@
+    // initialized, and we're also assuming a mint _and_ metadata have been
+    // created for the member.
+    // Collection memebers in this example can have their own separate
+    // metadata that differs from the metadata of the collection.
+    let account_info_iter = &mut accounts.iter();
+
+    // Accounts expected by this instruction:
+    //
+    //   0. `[w]`   Collection Member (Member)
+    //   1. `[]`    Collection Member (Member) Mint
+    //   2. `[s]`   Collection Member (Member) Mint authority
+    //   3. `[w]`   Collection (Group)
+    //   4. `[s]`   Collection (Group) update authority
+    let member_info = next_account_info(account_info_iter)?;
+    let member_mint_info = next_account_info(account_info_iter)?;
+    let member_mint_authority_info = next_account_info(account_info_iter)?;
+    let collection_info = next_account_info(account_info_iter)?;
+    let collection_update_authority_info = next_account_info(account_info_iter)?;
+
+    // Mint checks on the member
+    {
+        // IMPORTANT: this example program is designed to work with any
+        // program that implements the SPL token interface, so there is no
+        // ownership check on the mint account.
+        let member_mint_data = member_mint_info.try_borrow_data()?;
+        let member_mint = StateWithExtensions::<Mint>::unpack(&member_mint_data)?;
+
+        if !member_mint_authority_info.is_signer {
+            return Err(ProgramError::MissingRequiredSignature);
+        }
+        if member_mint.base.mint_authority.as_ref() != COption::Some(member_mint_authority_info.key)
+        {
+            return Err(TokenGroupError::IncorrectAuthority.into());
+        }
+    }
+
+    // Increment the size of the collection
+    let mut buffer = collection_info.try_borrow_mut_data()?;
+    let mut state = TlvStateMut::unpack(&mut buffer)?;
+    let collection = state.get_first_value_mut::<Group>()?;
+
+    check_update_authority(
+        collection_update_authority_info,
+        &collection.update_authority,
+    )?;
+    let member_number = collection.increment_size()?;
+
+    // Allocate a TLV entry for the space and write it in
+    let mut buffer = member_info.try_borrow_mut_data()?;
+    let mut state = TlvStateMut::unpack(&mut buffer)?;
+    let (member, _) = state.init_value::<Member>(false)?;
+    *member = Member::new(*collection_info.key, member_number.into());
+
+    Ok(())
+}
+
+/// Processes an `SplTokenGroupInstruction`
+pub fn process(program_id: &Pubkey, accounts: &[AccountInfo], input: &[u8]) -> ProgramResult {
+    let instruction = TokenGroupInterfaceInstruction::unpack(input)?;
+    match instruction {
+        TokenGroupInterfaceInstruction::InitializeGroup(data) => {
+            msg!("Instruction: InitializeCollection");
+            process_initialize_collection(program_id, accounts, data)
+        }
+        TokenGroupInterfaceInstruction::UpdateGroupMaxSize(data) => {
+            msg!("Instruction: UpdateCollectionMaxSize");
```

Comment:
I did this intentionally since I was considering this an example program that serves the purpose of "collections" (ie. Joe's Collection Program), so even though I’m using the interface instructions, they are specifically tied to collection assets that my program specializes in.